sources/packages/backend/apps/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -200,14 +200,14 @@ export class AssessmentControllerService {
     disbursementSchedules: DisbursementSchedule[],
   ): number {
     return disbursementSchedules
-      .flatMap(
-        (disbursementSchedule) => disbursementSchedule.disbursementValues,
+      .flatMap((disbursementSchedule) =>
```

Comment:
I would advise to use the filter after the flatmap instead of using in each iteration. like this. 

```ts
return disbursementSchedules
      .flatMap(
        (disbursementSchedule) => disbursementSchedule.disbursementValues,
      )
      .filter(
        (disbursementValue) =>
          disbursementValue.valueType !== DisbursementValueType.BCTotalGrant,
      )
      .reduce(
        (accumulator, disbursementValue) =>
          accumulator + disbursementValue.valueAmount,
        0,
      );
```